### PR TITLE
#1322 redesign details section

### LINKED ISSUE #1132: Finance - Edit Reimbursement Request
### Description

We need an endpoint for editing a reimbursement request

### Acceptance Criteria

- [x] New Post Receiver in the reimbursement routes file
  - Receiver will not have all the required parts for editing a reimbursement request
  - Recepient - User
  - Purchased From - Vendor
  - Refund Source - Account
  - Expense Type - Expense Type
  - Total Cost - int min zero
  - Date of expense - date
  - all reimbursement products - reimbursement products[]
  - Route will be on /reimbursement-requests/:id/edit
- [x] New Controller function called editReimbursementRequest in the reimbursement controller file
  - Parses the id out of the query
  - Gets the current user
  - Calls Service function 
- [x] New service function called editReimbursementRequest in reimbursement services file
  - Function will throw if user is not the original submitter or the request has already been approved by finance
  - throws if deleted
  - throws if not correct id
  - updates the reimbursement request with the given information

- [x] Unit Tests
### Proposed Solution

look at edit work package for guidance

### Mocks

_No response_

### LINKED ISSUE #1322: Project Form Redesign - Redesign Details Section
### Description

Redesign the Details Section so that it looks in accordance with the new design. 

### Acceptance Criteria

Looks like Mockup

### Proposed Solution

will definitely remove the pageblock. would suggest using a grid layout for it so its easy to manage smaller screens. Id use project form labels for the headers. Also team is acutally Teams and should be an array of teams. 

Work in the file Project Edit Details, but rename the component to be ProjectFormDetails and rename the file to be the same. 

Work off of the feature/Project-Form-Redesign Feature Branch

### Mocks

Old Version
Screen Shot 2023-09-05 at 6 14 48 PM

New Version
Screen Shot 2023-09-05 at 6 14 32 PM

### PR DESCRIPTION
**Pull request by Aaryan1203**
Reformatted project edit details section
Created a teams dropdown form

Change Request dropdown no longer works

## Screenshots
image

Smaller Screen:
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1322

### CODE REVIEW

**Review by ninest — CHANGES_REQUESTED** [review]
Can you make it responsive by having one field per row on smaller screens?

**Comment by ninest** [comment]
@Aaryan1203 Can you update the screenshots?

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 1]
```diff
@@ -41,7 +45,28 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({
             />
           </FormControl>
         </Grid>
-        <Grid item xs={4}>
+        <Grid item lg={2.4} xs={12}>
+          <FormControl fullWidth>
+            <ChangeRequestDropdown control={control} name="crId" errors={errors} />
```
Comment (ninest) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:50]: @RChandler234 Should the TeamsDropdown component also take in errors as a prop and display them if required?

**Review by ninest — DISMISSED** [review]

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx (hunk 1 above)
Comment (RChandler234) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:50]: yes

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 2]
```diff
@@ -65,7 +105,7 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({
             value={userToAutocompleteOption(users.find((user) => user.userId.toString() === projectLead))}
           />
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+        <Grid item xs={12} md={3} sx={{ mt: 1 }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:108]: these two fields should look like the rest at this screen size 
image

**Review by RChandler234 — CHANGES_REQUESTED** [review]
should not be any yarn.lock changes in this PR

**Comment by Aaryan1203** [comment]
Closed because it was on the wrong branch. Created a new PR.

**Pull request closed without merging**